app/components/primer/component.rb

```diff
@@ -42,5 +42,47 @@ def validate_aria_label
     def silence_deprecations?
       Rails.application.config.primer_view_components.silence_deprecations
     end
+
+    def check_no_user_tag(**arguments)
+      check_denylist({ [:tag] => "This component has a fixed tag." }, **arguments)
```

Comment:
since this is super straightforward, I wonder if we should do something like we do for `class`

```rb
raise ArgumentError, "This component has a fixed tag." if arguments.key?(:tag) && raise_on_invalid_options? && !ENV["PRIMER_WARNINGS_DISABLED"]
```

Reply:
👍 Yeah, that's much simpler!